tests/mocks/repository/lib/mock_mpool.c

```diff
@@ -291,6 +293,9 @@ mblock_rw(u64 id, const struct iovec *iov, int niov, size_t off, bool read)
         off += iov[i].iov_len;
     }
 
+    if (!read)
```

Comment:
I added this because it seemed like it was missing. Check my worl.

Reply:
There's also:

if (!PAGE_ALIGNED(iov[i].iov_base)) {